File: testsuite/matlab_and_octave/util/convert_features_and_add_preproc.m
function y = convert_features_and_add_preproc()
	global order;
	global gap;
	global reverse;
	global feature_type;
	y=false;

	if isempty(order)
		return;
	end

	if strcmp(feature_type, 'Ulong')==1
		type='ULONG';
	elseif strcmp(feature_type, 'Word')==1
		type='WORD';
	else
		return;
	end

	sg('add_preproc', strcat('SORT', type, 'STRING'));
	sg('convert', 'TRAIN', 'STRING', 'CHAR', 'STRING', type, ...
		order, order-1, gap, reverse);
	sg('attach_preproc', 'TRAIN');

	sg('convert', 'TEST', 'STRING', 'CHAR', 'STRING', type, ...
		order, order-1, gap, reverse);
	sg('attach_preproc', 'TEST');

	y=true;

File: testsuite/r/util/convert_features_and_add_preproc.R
convert_features_and_add_preproc <- function() {
	if (!exists('order', mode='numeric')) {
		return(FALSE)
	}

	if (regexpr('Ulong', feature_type)>0) {
		type <- 'ULONG'
	} else if (regexpr('Word', feature_type)>0) {
		type='WORD'
	} else {
		return(FALSE)
	}

	sg('add_preproc', paste('SORT', type, 'STRING', sep=''))
	sg('convert', 'TRAIN', 'STRING', 'CHAR', 'STRING', type,
		order, order-1, gap, reverse)
	sg('attach_preproc', 'TRAIN')

	sg('convert', 'TEST', 'STRING', 'CHAR', 'STRING', type,
		order, order-1, gap, reverse)
	sg('attach_preproc', 'TEST')

	return(TRUE)
}
